### ISSUE TITLE
Bridge for views

### ISSUE THREAD

**Issue by remq** [livefront/bridge#10]
I'm trying to implement Bridge for the app I'm building. Currently I use Icepick, which has an implementation for views defined in their documentation. Can I also use Bridge for these view instance state implementations?

**Comment by byencho** [comment]
That's a good question. There is currently no support for using `Bridge` with views in the way `Icepick` is set up to do it. Part of the reason for this is that it seemed unlikely that state saving in custom views would have the same problem that you'd find in Activities and Fragments, where you are much more likely to try putting your "network" data in the saved state Bundle. I'd recommend continuing to use `Icepick` in your views and if you actually see problems there we could look into adding support for it.

**Comment by remq** [comment]
Thanks for the response! Currently my custom view is setup pretty poorly: the data model for the list it's displaying is managed inside the view code. This model can get quite large (the list has infinite scroll), so I was hoping to store it using `Bridge`. I've already got some crashes coming in, but I guess now is a good time to refactor my code then ;)

**Comment by byencho** [comment]
Yeah, moving that code out of the view is probably a good idea anyway. Or even just offloading the state saving and restoration to the parent Activity / Fragment would work.

**Comment by byencho** [comment]
@remq I'm going to close this issue for now. If there ends up being more demand for this feature in the future we can reopen it.

**Comment by msecilmis** [comment]
Is there a progress on this? We actually need this for a couple of custom views.

**Comment by byencho** [comment]
Sorry, not yet. As a workaround you can try pulling out that data from your `View`, saving it with `Bridge` in the `Activity` / `Fragment` hosting it, and then putting that data back in manually. But I'm still hoping to get around to adding support for this.

**Comment by byencho** [comment]
@msecilmis You can also maybe toy around with something like this in the `View`:

```java
@Override
protected Parcelable onSaveInstanceState() {
    Bundle bundle = new Bundle();
    Bridge.saveInstanceState(this, bundle);
    bundle.putParcelable("parentState", super.onSaveInstanceState());
    return bundle;
}

@Override
protected void onRestoreInstanceState(Parcelable state) {
    Bundle bundle = (Bundle) state;
    Bridge.restoreInstanceState(this, bundle);
    Parcelable parentState = bundle.getParcelable("parentState");
    super.onRestoreInstanceState(parentState);
}
```

I haven't actually tested it so I can't guarantee that works but that's the gist of what I'd be doing internally once I add support for this.

**Comment by byencho** [comment]
Reopening this issue for better tracking.

**Comment by byencho** [comment]
@msecilmis This feature has been released in [v1.2.0](https://github.com/livefront/bridge/releases/tag/v1.2.0). Let me know if you have any questions
